src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -720,36 +722,30 @@ private static Expression BindParameterFromBody(ParameterInfo parameter, bool al
             factoryContext.JsonRequestBodyType = parameter.ParameterType;
             factoryContext.AllowEmptyRequestBody = allowEmpty || isOptional;
 
-            var convertedBodyValue = Expression.Convert(BodyValueExpr, parameter.ParameterType);
-            var argument = Expression.Variable(parameter.ParameterType, $"{parameter.Name}_local");
-
             if (!factoryContext.AllowEmptyRequestBody)
             {
                 // If the parameter is required or the user has not explicitly
                 // set allowBody to be empty then validate that it is required.
                 //
-                // Todo body_local = Convert(bodyValue, ToDo);
-                // if (body_local == null)
+                // if (bodyValue == null)
                 // {
                 //      wasParamCheckFailure = true;
                 //      Log.RequiredParameterNotProvided(httpContext, "Todo", "body");
                 // }
                 var checkRequiredBodyBlock = Expression.Block(
-                    Expression.Assign(argument, convertedBodyValue),
                     Expression.IfThen(
-                    Expression.Equal(argument, Expression.Constant(null)),
+                    Expression.Equal(BodyValueExpr, Expression.Constant(null)),
```

Comment:
This is a fix. We don't cast from `object` to the struct type before checking `bodyValue == null`.